### PR TITLE
Kernel: More OOM handling, and start fixing allocation under spinlock. 

### PR DESCRIPTION
**Pull request by bgianfo**
This series makes some minor progress on OOM handling (#6369).
It also starts working on fixing allocations performed while holding spinlocks, found with `KMALLOC_VERIFY_NO_SPINLOCK_HELD` introduced in  9c384756088658e02ee6fba10030efcce1dc876b.

### CODE REVIEW

File: Kernel/ProcessGroup.cpp [hunk 1]
```diff
@@ -30,10 +30,12 @@ RefPtr<ProcessGroup> ProcessGroup::create(ProcessGroupID pgid)
...
+    {
+        ScopedSpinLock lock(g_process_groups_lock);
 
-    if (auto existing = from_pgid(pgid))
-        return existing.release_nonnull();
+        if (auto existing = from_pgid(pgid))
+            return existing;
+    }
 
     return create(pgid);
```
Comment (awesomekling) [inline Kernel/ProcessGroup.cpp:40]: Hold up, isn't this racy if two threads are calling `ProcessGroup::find_or_create()` at the same time? Both of them could end up in `create()`, creating two of the same `ProcessGroup` and putting both in the global list.

File: Kernel/Arch/i386/CPU.cpp [hunk 2]
```diff
@@ -920,15 +920,13 @@ UNMAP_AFTER_INIT void write_xcr0(u64 value)
 
 READONLY_AFTER_INIT FPUState Processor::s_clean_fpu_state;
```
Comment (awesomekling) [inline Kernel/Arch/i386/CPU.cpp:922]: Typo in commit message, Their's -> There's.

File: Kernel/ProcessGroup.cpp (hunk 1 above)
Comment (bgianfo) [reply Kernel/ProcessGroup.cpp]: True, reworked this to avoid the race.

File: Kernel/Arch/i386/CPU.cpp (hunk 2 above)
Comment (bgianfo) [reply Kernel/Arch/i386/CPU.cpp:922]: Yup, fixed, thanks!

**Pull request merged**